Providers/BotInfoProvider.cs

```diff
@@ -0,0 +1,30 @@
+using System.Collections.Generic;
+using Microsoft.Extensions.Configuration;
+
+namespace Theorem.Providers
+{
+    /// <summary>
+    /// SlackProvider provides all Slack functionality (send/receive/etc)
+    /// </summary>
+    public class BotInfoProvider
```

Comment:
I intentionally used the BotInfoProvider to be able to provide the list of Middleware stripped from all unnecessary type information. Also I have further ideas the BotInfoProvider might be useful for that I'd like to implement. I can change it though if you want